DigitalLearningSolutions.Web/Views/MyAccount/_EditRegistrationNumber.cshtml

```diff
@@ -4,23 +4,23 @@
 
 @{
   var errorHasOccurred = !ViewData.ModelState.IsValid;
-  var optionNoneSelected =  @Model.HasProfessionalRegistrationNumber == YesNoSelectionEnum.None;
-  var optionYesSelected = @Model.HasProfessionalRegistrationNumber == YesNoSelectionEnum.Yes;
-  var optionNoSelected = @Model.HasProfessionalRegistrationNumber == YesNoSelectionEnum.No;
+  var optionNoneSelected = Model.HasProfessionalRegistrationNumber == YesNoSelectionEnum.None;
+  var optionYesSelected = Model.HasProfessionalRegistrationNumber == YesNoSelectionEnum.Yes;
+  var optionNoSelected = Model.HasProfessionalRegistrationNumber == YesNoSelectionEnum.No;
   var professionalRegConditionalId = "professional-reg-conditional-Id";
 }
 
-<div class="nhsuk-form-group @(optionNoneSelected? "nhsuk-form-group--error": "")">
+<div class="nhsuk-form-group @(errorHasOccurred && optionNoneSelected ? "nhsuk-form-group--error" : "")">
   <fieldset class="nhsuk-fieldset" aria-describedby="professional-registration-hint">
     <div class="nhsuk-hint" id="professional-registration-hint">
       Do you have a Professional Registration Number?
     </div>
 
-  @if (errorHasOccurred && optionNoneSelected) {
-    <span class="nhsuk-error-message" id="no-selection-error">
-       @(!optionYesSelected? "Select an option":"")
-    </span>
-  }
+    @if (errorHasOccurred) {

```

Comment:
This currently means this error section will show up if there's an error anywhere on the ViewModel, but if we change errorHasOccurred like I've described above we should get around that.

Reply:
This error section will now only be shown when there is an error on HasProfessionalRegistrationNumber property